### PR TITLE
Implement toQueryString to convert key/value pair data to query string

### PR DESCRIPTION
**Pull request by lchanmann**

### CODE REVIEW

**Review by jlhughes — COMMENTED** [review]
Where/how is this being used?

**Comment by lchanmann** [comment]
I'm planning to use it for constructing http POST data in query string format https://github.com/nanoPayinc/NANOPAY/blob/POC_accounting_entry/nanopay/src/net/nanopay/accounting2/ServerAccountingService.js#L114-L125.

File: src/foam/util/StringUtil.java [hunk 1]
```diff
@@ -118,4 +118,18 @@ public static String setSuffix(String s, String suffix, boolean ignoreCase) {
       return s;
     return s + suffix;
   }
+
+  public static String toQueryString(String... data) {
+    if ( data.length % 2 == 1 )
+      throw new RuntimeException("Invalid query string data");
+
+    var sb = new StringBuilder();
+    for ( var i = 0; i < data.length - 1; i+=2 ) {
```
Comment (kgrgreer) [inline src/foam/util/StringUtil.java]: spaces before the ;'s and around the +=

File: src/foam/util/StringUtil.java [hunk 2]
```diff
@@ -118,4 +118,18 @@ public static String setSuffix(String s, String suffix, boolean ignoreCase) {
...
+      if ( i > 0 )
+        sb.append('&');
+
+      sb.append(data[i]).append('=').append(data[i+1]);
```
Comment (kgrgreer) [inline src/foam/util/StringUtil.java]: Us any escaping needed?

**Pull request merged**